src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -125,17 +134,32 @@ public Response forward(@NotNull @Valid RestAction ra) {
 
         Action action = builder.build();
         String serializedAction = Parser.encode(action);
-        emitter.send(buildMessageWithId(serializedAction));
-        forwardedActions.increment();
 
-        return Response.ok().build();
+        CompletableFuture<Void> callback = new CompletableFuture<>();
+        emitter.send(buildMessageWithId(serializedAction, callback));
+        try {
+            callback.get(callbackTimeout, TimeUnit.SECONDS);
+            forwardedActions.increment();
+            return Response.ok().build();
+        } catch (Exception e) {
+            Log.error("Message delivery to Kafka failed", e);
+            return Response.status(SERVICE_UNAVAILABLE).build();
+        }
     }
 
-    private static Message<String> buildMessageWithId(String payload) {
+    private static Message<String> buildMessageWithId(String payload, CompletableFuture<Void> callback) {
         byte[] messageId = UUID.randomUUID().toString().getBytes(UTF_8);
-        OutgoingKafkaRecordMetadata metadata = OutgoingKafkaRecordMetadata.builder()
+        OutgoingKafkaRecordMetadata<String> metadata = OutgoingKafkaRecordMetadata.<String>builder()
                 .withHeaders(new RecordHeaders().add(MESSAGE_ID_HEADER, messageId))
                 .build();
-        return Message.of(payload).addMetadata(metadata);
+        return Message.of(payload)
+                .addMetadata(metadata)
+                .withAck(() -> {
+                    callback.complete(null);
+                    return CompletableFuture.completedFuture(null);
+                }).withNack(reason -> {
+                    callback.completeExceptionally(reason);
+                    return CompletableFuture.completedFuture(null);
+                });
```

Comment:
As weird as it seems, this is [inspired from SmallRye Reactive Messaging itself](https://github.com/smallrye/smallrye-reactive-messaging/blob/e59e7cff9962c0cdf4a7f6de859add786a2a6a23/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/extension/EmitterImpl.java#L30-L38).